plugin/storage/badger/factory.go

```diff
@@ -77,18 +77,18 @@ type Factory struct {
 // NewFactory creates a new Factory.
 func NewFactory() *Factory {
 	return &Factory{
-		Options:         NewOptions(),
+		Config:          NewConfig(),
 		maintenanceDone: make(chan bool),
 	}
 }
 
 func NewFactoryWithConfig(
-	cfg NamespaceConfig,
+	cfg Config,
 	metricsFactory metrics.Factory,
 	logger *zap.Logger,
 ) (*Factory, error) {
 	f := NewFactory()
-	f.configureFromOptions(&Options{Primary: cfg})
+	f.configureFromConfig(&cfg)
```

Comment:
can simpler name work?
```suggestion
	f.configure(&cfg)
```